paasta_tools/contrib/get_running_task_allocation.py

```diff
@@ -143,14 +143,17 @@ def get_kubernetes_metadata(
     Optional[str],
     Optional[str],
 ]:
+    node = kubernetes_tools.get_pod_node(kube_client, pod, cache_nodes=True)
+    if node:
+        pool = node.metadata.labels.get("yelp.com/pool", "default")
+    else:
+        pool = "default"
```

Comment:
Maybe it would be worth pulling this out into a function?

I think this needs to handle the case where `labels` is None, just in case there are some rogue nodes. Also, what do you think also falling back to the node selector?

Reply:
Nice, yeah I think it would be cleaner as a separate function and good shout on the labels being optional. Not so sure it's worth having a fallback though, I'm happy to add it if you feel it's too risky without but it feels like extra complexity for small reassurance.